src/main/java/org/folio/circulation/domain/notice/schedule/ScheduledDigitalReminderHandler.java

```diff
@@ -206,22 +248,40 @@ private CompletableFuture<Result<ScheduledNoticeContext>> createFeeFineAction(Sc
    */
   @Override
   protected CompletableFuture<Result<ScheduledNotice>> updateNotice(ScheduledNoticeContext context) {
-    OverdueFinePolicyRemindersPolicy.ReminderSequenceEntry nextReminder = context.getLoan().getNextReminder();
+    RemindersPolicy.ReminderConfig nextReminder = context.getLoan().getNextReminder();
     if (nextReminder == null) {
       return deleteNotice(context.getNotice(), "no more reminders scheduled");
     } else if (isNoticeIrrelevant(context)) {
       return deleteNotice(context.getNotice(), "further reminder notices became irrelevant");
     } else {
-      ScheduledNotice nextReminderNotice = context.getNotice()
-        .withNextRunTime(nextReminder.getPeriod().plusDate(systemTime));
-      nextReminderNotice.getConfiguration()
-        .setTemplateId(nextReminder.getNoticeTemplateId())
-        .setFormat(nextReminder.getNoticeFormat());
-
-      return scheduledNoticesRepository.update(nextReminderNotice);
+      return buildNextNotice(context, nextReminder)
+        .thenCompose(scheduledNoticeResult -> scheduledNoticesRepository.update(scheduledNoticeResult.value()));
     }
   }
 
+  protected CompletableFuture<Result<ScheduledNotice>> buildNextNotice (
+    ScheduledNoticeContext context, RemindersPolicy.ReminderConfig nextReminder) {
+
+    return configurationRepository.findTimeZoneConfiguration()
+      .thenCompose(tenantTimeZone ->
+        nextReminder
+          .nextNoticeDueOn(
+            systemTime,
+            tenantTimeZone.value(),
+            context.getLoan().getCheckoutServicePointId(),
+            calendarRepository
```

Comment:
In mod-circulation we usually don't start each parameter from a new line, we use 'natural flow' instead. I mean, typically we would format it like this
```
.thenCompose(tenantTimeZone -> nextReminder.nextNoticeDueOn(systemTime,
  tenantTimeZone.value(), context.getLoan().getCheckoutServicePointId(),
  calendarRepository)
```
There are many places like this in the PR, so I'll just leave this one comment instead of commenting every case